execute.go

```diff
@@ -285,8 +285,25 @@ func max(a, b int) int {
 	return b
 }
 
+func findSelection(matchString string, selectionSet ast.SelectionSet, fragmentDefs ast.FragmentDefinitionList) (*ast.Field, error) {
```

Comment:
It was a bit strange to see `ApplyFragments` and `SelectedFields` used like this since they do try to solve very similar problems. I wonder if the code wouldn't be simpler if you just used `ApplyFragments` and then did a type assertion to `Field` for the filter

Reply:
That would be ideal! I tried doing that but that made an existing test (`TestExecutor_insertIntoLists`) fail. It looks like for that one test, doing `ApplyFragments` is not enough to find the selection for `friends`. I'm trying to look into it, but that's why I decided to write this if/else.

Reply:
Huh that is weird 🤔 Well hopefully that isn't too tricky to hunt down